src/ContentBlocks/BlockButton/BlockButton.js

```diff
@@ -21,14 +22,14 @@ export default function BlockButton({
   assetType,
 }: Props) {
   const isPrimary = primary === 'True';
-  let props = { label, primary: isPrimary, target: '_blank' };
-  if (assetType === 'path' && path && path.indexOf('.') < 0) {
-    props = { ...props, path };
-  } else if (assetType === 'path' && path && path.indexOf('.') > -1) {
-    props = { ...props, href: path, target: '_blank' };
-  } else {
-    props = { ...props, href };
-  }
+  const link = getLink(assetType, path, href);
+  const tracking = {
+    'data-analytics-track': 'true',
+    'data-analytics-value': assetType === 'path' ? path : href,
+    'data-analytics-type': getAnalyticsType(assetType, path, href),
+    'data-analytics-category': buttonType,
+  };
+  const props = { label, primary: isPrimary, target: '_blank', ...tracking, ...link };
```

Comment:
Wouldn't this always open in a blank tab? Internal links need to stay in the same window.